fhir-model/src/main/java/com/ibm/fhir/model/path/FHIRPathQuantityNode.java

```diff
@@ -168,24 +172,59 @@ public int compareTo(FHIRPathNode other) {
             throw new IllegalArgumentException();
         }
         if (hasValue()) {
-            if (other instanceof FHIRPathQuantityValue) {
-                return getValue().compareTo(other);
+            if (other instanceof FHIRPathQuantityValue || other.getValue() instanceof FHIRPathQuantityValue) {
+                FHIRPathQuantityValue quantityValue = (other instanceof FHIRPathQuantityValue) ? (FHIRPathQuantityValue) other : (FHIRPathQuantityValue) other.getValue();
+                return getValue().compareTo(quantityValue);
             }
-            if (other.getValue() instanceof FHIRPathQuantityValue) {
-                return getValue().compareTo(other.getValue());
+            if (other instanceof FHIRPathNumberValue || other.getValue() instanceof FHIRPathNumberValue) {
+                FHIRPathNumberValue numberValue = (other instanceof FHIRPathNumberValue) ? (FHIRPathNumberValue) other : (FHIRPathNumberValue) other.getValue();
+                return getValue().compareTo(numberValue);
             }
-        } else if (other instanceof FHIRPathQuantityNode) {
+        }
+        if (other instanceof FHIRPathQuantityNode) {
             return getQuantityValue().compareTo(((FHIRPathQuantityNode) other).getQuantityValue());
-        } else if (other instanceof FHIRPathNumberValue) {
-            return getQuantityValue().compareTo(((FHIRPathNumberValue) other).decimal());
-        } else if (other.getValue() instanceof FHIRPathNumberValue) {
-            return getQuantityValue().compareTo(((FHIRPathNumberValue) other.getValue()).decimal());
         }
-        throw new IllegalStateException();
+        FHIRPathNumberValue numberValue = (other instanceof FHIRPathNumberValue) ? (FHIRPathNumberValue) other : (FHIRPathNumberValue) other.getValue();
+        return getQuantityValue().compareTo(numberValue.decimal());
+    }
+    
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof FHIRPathNode)) {
+            return false;
+        }
+        FHIRPathNode other = (FHIRPathNode) obj;
+        if (!isComparableTo(other)) {
```

Comment:
Block 207 on repeats in `isComparableTo`

Reply:
isComparableTo check to see whether we can compare. The block in equals checks to see if compareTo result == 0.